index.d.ts

```diff
@@ -682,10 +686,57 @@ export interface WithT {
   t: TFunction;
 }
 
-export type TFunctionResult = string | object | Array<string | object> | undefined | null;
+/**
+ * Object returned from t() function when passed returnDetails: true option.
+ */
+export type TFunctionResolvedResult<T = string> = {
```

Comment:
Rename to `TFunctionDetailsResult`?

Reply:
`TFunctionDetailedResult` would be better…

Reply:
done